binance_smart_chain/dex/trades/insert_dodo.sql

```diff
@@ -0,0 +1,312 @@
+
+        UNION ALL
+
+        -- dodov2 dpp
+        SELECT
+            evt_block_time AS block_time,
+            'DODOEX' AS project,
+            'v2_bsc' AS version,
+            'DEX' AS category,
+            trader AS trader_a,
+            receiver AS trader_b,
+            "fromAmount" AS token_a_amount_raw,
+            "toAmount" AS token_b_amount_raw,
+            NULL::numeric AS usd_amount,
+            "fromToken" AS token_a_address,
+            "toToken" AS token_b_address,
+            contract_address AS exchange_contract_address,
+            evt_tx_hash AS tx_hash,
+            NULL::integer[] AS trace_address,
+            evt_index
+        FROM
+            dodoex."DPP_evt_DODOSwap"
+        WHERE trader <> '\xbe60d4c4250438344bec816ec2dec99925deb4c7'
+
+        UNION ALL
+
+        -- dodov2 dsp
+        SELECT
+            evt_block_time AS block_time,
+            'DODOEX' AS project,
+            'v2_bsc' AS version,
+            'DEX' AS category,
+            trader AS trader_a,
+            receiver AS trader_b,
+            "fromAmount" AS token_a_amount_raw,
+            "toAmount" AS token_b_amount_raw,
+            NULL::numeric AS usd_amount,
+            "fromToken" AS token_a_address,
+            "toToken" AS token_b_address,
+            contract_address AS exchange_contract_address,
+            evt_tx_hash AS tx_hash,
+            NULL::integer[] AS trace_address,
+            evt_index
+        FROM
+            dodoex."DSP_evt_DODOSwap"
+        WHERE trader <> '\xbe60d4c4250438344bec816ec2dec99925deb4c7'
+    ) dexs
+    INNER JOIN bsc.transactions tx
+        ON dexs.tx_hash = tx.hash
+        AND tx.block_time >= start_ts
+        AND tx.block_time < end_ts
+        AND tx.block_number >= start_block
+        AND tx.block_number < end_block
+    LEFT JOIN bep20.tokens erc20a ON erc20a.contract_address = dexs.token_a_address
+    LEFT JOIN bep20.tokens erc20b ON erc20b.contract_address = dexs.token_b_address
+    LEFT JOIN prices.usd pa ON pa.minute = date_trunc('minute', dexs.block_time)
+        AND pa.contract_address = dexs.token_a_address
+        AND pa.minute >= start_ts
+        AND pa.minute < end_ts
+    LEFT JOIN prices.usd pb ON pb.minute = date_trunc('minute', dexs.block_time)
+        AND pb.contract_address = dexs.token_b_address
+        AND pb.minute >= start_ts
+        AND pb.minute < end_ts
+    -- LEFT JOIN prices.layer1_usd pe ON pe.minute = date_trunc('minute', dexs.block_time)
+    --     AND pe.symbol = 'BNB'
+    LEFT JOIN prices.usd pe ON pe.minute = date_trunc('minute', dexs.block_time)
+        AND pe.symbol = 'WBNB'
+        AND pe.minute >= start_ts
+        AND pe.minute < end_ts
+    WHERE dexs.block_time >= start_ts
+    AND dexs.block_time < end_ts
+    AND dexs.token_a_address <> dexs.token_b_address
+
+    ON CONFLICT DO NOTHING
+    RETURNING 1
+)
+SELECT count(*) INTO r from rows;
+RETURN r;
+END
+$function$;
+
+-- fill 2020
+SELECT dex.insert_dodo(
```

Comment:
please edit function call to correct name: dex.insert_dodoex_bsc

this applies to all historical calls below -- cron job seems to be correct